IOPool/Streamer/interface/StreamerInputFile.h

```diff
@@ -51,7 +54,7 @@ namespace edm {
 
   private:
     void openStreamerFile(std::string const& name, std::string const& LFN);
-    storage::IOSize readBytes(char* buf, storage::IOSize nBytes);
+    storage::IOSize readBytes(char** buf, storage::IOSize nBytes, bool zeroCopy = true, unsigned skipped = 0);
```

Comment:
I think using `char*& buf` would be cleaner as it better expresses what is being done.

Reply:
Looking further at the code, I actually think the better solution would be to change the signature to return the starting offset, 
```
std::pair<storage::IOSize, char*> readBytes(char* buf, storage::IOSize nBytes, bool zeroCopy = true, unsigned skipped = 0)
```

That way you don't have to worry about accidentally modifying the address of the buffer which seems to be why you were forced not no longer call using the argument `&eventBuf_[sizeof(EventHeader)]`

Reply:
What complicated it here it is that the header is read first, so the offset moves by that amount, then creation of the message view in the end needs everything from the start of the header.
`&eventBuf_[sizeof(EventHeader)]` is a single pointer (and not a variable), so it can't work here. Instead I pass **buf which points to the beginning of the small buffer. If there is no zero-copy, I copy payload to the buffer at offset skippedHdr. If there is zero-copy, I overwrite the pointer value to the position of big buffer offset - skippedHdr (so it starts at the header).

As you suggest, we could areturn the final pointer in a pair and then call xxPtr = pair.second, instead of passing `char **` or `char *&` to the function.

I just realized there may be another problem - resize can reallocate vector buffer in heap, so `headerPtr = &headerBuf_[0];` and `eventPtr = &eventBuf_[0];` should be called again after resize (of the small buffer).